### ISSUE TITLE
CentOS Stream CoreOS 417 Boot Fails

### ISSUE THREAD

**Issue by necouchman** [okd-project/okd#2077]
**Describe the bug**
I'm attempting to install OKD 4.17 in the user-provisioned manner. I've downloaded the recommended FCOS (39.20231101.3.0) and am booting into the ISO of that, and then running the coreos-installer with the ignition files generated by the installer. When I run the coreos-installer with the bootstrap.ign file, the system reboots and then I get a couple of interesting items:
* The grub loader says "CentOS Stream CoreOS 417...", which is different from the FCOS that I installed.
* If I try to boot into the CentOS Stream CoreOS 417 option, I get an error: bad shim signature.
* If I boot into the second option, "Fedora CoreOS 39...", the system boots, but the bootstrap services don't actually start correctly.

![image](https://github.com/user-attachments/assets/66399d07-7d81-43d1-a344-7c5f810c2a09)

```
Dec 26 04:58:44 bootstrap.domain.local bootkube.sh[3221]: /usr/local/bin/bootkube.sh: line 85: oc: command not found
Dec 26 04:58:44 bootstrap.domain.local systemd[1]: bootkube.service: Main process exited, code=exited, status=127/n/a
Dec 26 04:58:44 bootstrap.domain.local systemd[1]: bootkube.service: Failed with result 'exit-code'.
```

So, I'm wondering a couple of things:
* What is this "CentOS Stream CoreOS"? I can't find any downloads for it or references to it outside of a few obscure places.
* If I'm provisioning the infrastructure myself, what CoreOS should I actually be using? Is FCOS 41 compatible? Or is there some other ISO I should be downloading to use with the installer-generated ignition files?

**Version**
[4.17.0-okd-scos.0](https://github.com/okd-project/okd/releases/tag/4.17.0-okd-scos.0)

**How reproducible**
100% of the time.

**Log bundle**
My cluster isn't even far enough up to produce this information:
```
$ ./openshift-install gather bootstrap --dir ./okd-sandbox/ --bootstrap 10.73.7.21
INFO Pulling Cluster API artifacts                
INFO Failed to gather Cluster API manifests: either Cluster API manifests not generated or terraform provision 
INFO Skipping VM console logs gather: no platform configured in metadata 
INFO Pulling debug logs from the bootstrap machine 
INFO Failed to gather bootstrap logs: failed to create SSH client: failed to use pre-existing agent, make sure the appropriate keys exist in the agent for authentication: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain 
FATAL failed to gather VM console and bootstrap logs 
```
Please let me what information I can provide.

**Comment by JaimeMagiera** [comment]
Hi,

Here are a couple resources to help familiarize you with the current state of affairs:

https://okd.io/blog/2024/06/01/okd-future-statement/
https://okd.io/docs/project/scos-migration-faq/

In short, there are no publicly available SCOS artifacts at this time. You'll want to use the FCOS image as noted. I believe the workaround for the issue you're seeing is to install oc on the nodes. We'll write something up in the new year. Most of us are currently on holiday. 

hope that helps.

Jaime

**Comment by necouchman** [comment]
Thanks @JaimeMagiera,
I've also found discussion #2015, which seems to be related; however, the work-around described in that discussion has to do with either Assisted or Agent Install modes, so it isn't a 1:1 match with a User-Provisioned Infrastructure install. I'm able to locate the Live ISO for RHCOS, so I'm going to give that a try, I'll also install oc on those systems and see if that works.

**Comment by titou10titou10** [comment]
> has to do with either Assisted or Agent Install modes, so it isn't a 1:1 match with a User-Provisioned Infrastructure install. 

There is some confusion here. "Assisted or Agent Install modes" is a way to install OKD with UPI..Like iPXE install (this requires an "extra" bootstrap node). ie you setup your nodes, then install OKD with ABI or AI or iPXE...
I'm the author of #2015, #2035 ...and always install OKD with ABI on proxmox..
Check https://github.com/okd-project/okd-web/issues/47 also

**Comment by necouchman** [comment]
Thanks @titou10titou10, appreciate the clarification on this. I'm a bit of a newbie to OKD, just getting my bearings, so I appreciate the guidance. I'm also install OKD on ProxMox, but I'm not using Assisted or Agent-Based installs, just generating the Ignition files and launching the installer myself.

**Comment by kai-uwe-rommel** [comment]
@necouchman when creating the UPI machines for the deployment, you need to disable secure boot for the machines. While FCOS can do secure boot, SCOS can't (missing shim signature).

**Comment by CrAazZyMaN21** [comment]
@necouchman I could install the cluster only with an older scos image (scos-413.9.202303020610-0-live.x86_64.iso) mentioned in (https://github.com/okd-project/okd-scos/issues/11). I extraced the vmlinuz, initrd.img and rootfs.img from the ISO and pxebooted with the Ignition Files from openshift-install. I used a dedicated boostrap vm and the RHCOS or FCOS images didnt work at all with UPI.

**Comment by kai-uwe-rommel** [comment]
@CrAazZyMaN21 strange report. I wonder what you did so the FCOS base for UPI installation failed. That is the current official way of deploying OKD, even with 4.17.

**Comment by GingerGeek** [comment]
Hi, just to confirm that Secure Boot currently needs to be disabled. This will likely be the case until we switch to a CentOS 10 based images (potentially OKD 4.20).

SCOS images are now available to download. These are baked into OKD's `openshift-installer` metadata and you can directly download the build artifacts here:
https://cloud.centos.org/centos/scos/9/prod/streams/latest/x86_64/

You should no longer need to "pivot" from Fedora

Are you still having an issue on the latest version of OKD when bootstrapping from SCOS images?

**Comment by kai-uwe-rommel** [comment]
> These are baked into OKD's openshift-installer metadata

@GingerGeek please explain. Starting with which OKD release?

**Comment by GingerGeek** [comment]
@kai-uwe-rommel It was introduced here:
https://github.com/openshift/installer/commit/ca251c3e4ac45543f80b65b05c0ccda2bfc3fc3d

and has been updated a few times since
e.g
https://github.com/openshift/installer/pull/9657


The installer is pulling from RedHat source rather than CentOS source currently.

@Prashanth684 may have more insight

**Comment by kai-uwe-rommel** [comment]
@GingerGeek I checked OKD [4.18.0-okd-scos.8](https://amd64.origin.releases.ci.openshift.org/releasestream/4-scos-stable/release/4.18.0-okd-scos.8) and it still specifies FCOS as the recommended launch OS.

**Comment by titou10titou10** [comment]
>  But apparently we can expect that to change to SCOS for 4.19, I read.

Yes. v4.19.0-okd-scos.ec.x embeds SCOS bootstrap images. Read [comments on this issue starting here](https://github.com/okd-project/okd/discussions/2015#discussioncomment-12605212).

**Comment by GingerGeek** [comment]
Apologies, I was clearly living ahead of where our releases were!!

I think if I was bootstrapping a bare metal 4.18 cluster today I would still use one of the SCOS images for initial boot and loading the ignition just because I feel it's a safer pivot but that's a gut instinct rather than something I've tested.

**Comment by kai-uwe-rommel** [comment]
Thanks. I do see that for the 4.19 ec.8 release the "openshift-install coreos print-stream-json" indeed lists SCOS while for 4.18 it still lists FCOS.

**Comment by imdmahajankanika** [comment]
> It is recommended for baremetal installations (UPI or ABI/AI) - to install the latest 4.19 ec using the artifacts here: https://cloud.centos.org/centos/scos/9/prod/streams/latest/x86_64/ . This will avoid issues like the `oc` binary not being present and has a bunch of fixes for ABI/AI as well.

Hello @Prashanth684 I tried to use raw image of scos for baremetal UPI for adding a new node, but during the boot processs it seems like it is working over ipv6 as it is not identifying the ipv4 DNS as shown below

<img width="1820" height="253" alt="Image" src="https://github.com/user-attachments/assets/b50a648c-a9be-433d-80ea-a59b031beb5c" />

Do you have any idea please?
